examples/ref-templates/assets/htlc.py

```diff
@@ -0,0 +1,100 @@
+"""This contract implements a "hash time lock".
+The contract will approve transactions spending algos from itself under two circumstances:
+
+- If an argument arg_0 is passed to the script such that TMPL_HASHFN(arg_0) is equal to TMPL_HASHIMG, 
+then funds may be closed out to TMPL_RCV.
+- If txn.FirstValid is greater than TMPL_TIMEOUT, then funds may be closed out to TMPL_OWN.
+
+The idea is that by knowing the preimage to TMPL_HASHIMG, funds may be released to 
+TMPL_RCV (Scenario 1). Alternatively, after some timeout round TMPL_TIMEOUT, 
+funds may be closed back to their original owner, TMPL_OWN (Scenario 2). 
+Note that Scenario 1 may occur up until Scenario 2 occurs, even if TMPL_TIMEOUT has already passed."""
+
+""" Parameters
+TMPL_RCV: the address to send funds to when the preimage is supplied
+TMPL_HASHFN: the specific hash function (sha256 or keccak256) to use (sha256 in this example)
+TMPL_HASHIMG: the image of the hash function for which knowing the preimage under TMPL_HASHFN will release funds
+TMPL_TIMEOUT: the round after which funds may be closed out to TMPL_OWN
+TMPL_OWN: the address to refund funds to on timeout
+TMPL_FEE: maximum fee of any transactions approved by this contract """
+
+from pyteal import *
+
+john = Addr("2UBZKFR6RCZL7R24ZG327VKPTPJUPFM6WTG7PJG2ZJLU234F5RGXFLTAKA")
+master = Addr("WWYNX3TKQYVEREVSW6QQP3SXSFOCE3SKUSEIVJ7YAGUPEACNI5UGI4DZCE")
+fee = 10000
+pre_image = "QzYhq9JlYbn2QdOMrhyxVlNtNjeyvyJc/I8d8VAGfGc="
+timeout = 2000
+
+def htlc(TMPL_RCV = john,
+         TMPL_OWN = master,
+         TMPL_FEE = fee,
+         TMPL_HASHIMG = pre_image,
+         TMPL_HASHFN = Sha256,
+         TMPL_TIMEOUT = timeout):
```

Comment:
Please move the comments to function doc string and remove default values. We will use the arguments in the function call:
```
def htlc(TMPL_RCV, TMPL_FEE, ... ):
    """HTLC template
    Parameters: 
      TMPL_RCV (string): the address to send funds to when the preimage is supplied
     ...
   """
```

and in the call:
```
htlc(john, ...)
```

Reply:
ok